frame/system/src/lib.rs

```diff
@@ -997,6 +1001,57 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+impl<T: Trait> Module<T> {
+	pub fn spent_weight() -> Option<Weight> {
+		CurrentDispatchableWeight::get()
+	}
+
+	/// Decrease the amount of weight consumed by the current dispatchable.
+	///
+	/// This function can only meaningfully called within `Self::dispatch`. Otherwise,
+	/// it does nothing and returns `Err`.
+	///
+	/// The total weight returned from the dispatchable cannot exceed its total requested weight.
+	pub fn return_unspent_weight(unspent: Weight) -> Result<(), ()> {
+		let weight = CurrentDispatchableWeight::take().ok_or(())?;
+		let next_weight = weight.checked_sub(unspent).ok_or(())?;
```

Comment:
Is Weight guaranteed to be always unsigned? Should we handle the case where `next_weight` becomes negative here?

Reply:
Yes weight is always unsigned and cannot be negative.